railties/lib/rails/railtie/forker.rb

```diff
@@ -0,0 +1,49 @@
+module Rails
+  class Forker
+    autoload :Runner, 'rails/railtie/forker/runner'
+
+
+    def runner=(runner)
+      @runner=runner 
+    end
+
+    def options
+      @options ||= {}
+    end
+
+    def options=(o)
+      options.merge!(o)
+    end
+
+    def before_fork(&blk)
+      @before_forks ||= []
+      @before_forks << blk if blk
+      @before_forks
+    end
+
+    def after_fork(&blk)
+      @after_forks ||= []
+      @after_forks << blk if blk
+      @after_forks
+    end
+
+
+    def fork!(label, app)
+      before_fork.each { |b| b.call(app) }
+      Process.fork do
+        $0 = "rails: #{label}"
+        after_fork.each { |a| a.call(app) }
+        begin
+          run!
+        rescue
+          exit
+        end
+      end
+
+    end
+
+    def run!
+      @runner::Runner.new(options).run!
```

Comment:
you were right, it was confusing. I've modified the config example in the description to adapt to recent changes: when setting ```forkers(Resque)```, it will automatically look for a ```Runner``` class defined within it. I hope it is clearer now.